invisible_cities/cities/isidora.py

```diff
@@ -0,0 +1,231 @@
+        """
+        n_events_tot = 0
+        # check the state of the machine
+        if not self.input_files:
+            raise IOError('input file list is empty')
+
+        if not self.setFiles:
+            raise IOError('must set files before running')
+
+        print("""
+                 ISIDORA will run a max of {} events
+                 Input Files ={}
+                          """.format(nmax, self.input_files))
+
+        # loop over input files
+        first = False
+        for ffile in self.input_files:
+            print("Opening", ffile, end="... ")
+            filename = ffile
+            try:
+                with tb.open_file(filename, "r+") as h5in:
+                    # access RWF
+                    pmtrwf  = h5in.root.RD.pmtrwf
+                    sipmrwf = h5in.root.RD.sipmrwf
+
+                    # Copy sensor table if exists (needed for GATE)
+                    if self.setCwfStore:
+                        if 'Sensors' in h5in.root:
+                            self.sensors_group = self.cwfFile.create_group(
+                                self.cwfFile.root, "Sensors")
+                            datapmt = h5in.root.Sensors.DataPMT
+                            datapmt.copy(newparent=self.sensors_group)
+
+                    if self.run_number > 0:
+                        self.eventsInfo = h5in.root.Run.events
+
+                    NEVT, NPMT, PMTWL   = pmtrwf.shape
+                    NEVT, NSIPM, SIPMWL = sipmrwf.shape
+                    print("Events in file = {}".format(NEVT))
+
+                    if first == False:
+                        print_configuration({"# PMT"  : NPMT,
+                                             "PMT WL" : PMTWL,
+                                             "# SiPM" : NSIPM,
+                                             "SIPM WL": SIPMWL})
+
+                        self.signal_t = np.arange(0., PMTWL * 25, 25)
+                        first = True
+                    # loop over all events in file unless reach nmax
+                    for evt in range(NEVT):
+                        # deconvolve
+                        CWF = blr.deconv_pmt(
+                          pmtrwf[evt],
+                          self.coeff_c,
+                          self.coeff_blr,
+                          n_baseline  = self.n_baseline,
+                          thr_trigger = self.thr_trigger)
+
+                        if self.setCwfStore:
+                            self.store_cwf(CWF)
+
+                        n_events_tot +=1
+                        if n_events_tot%self.nprint == 0:
+                            print('event in file = {}, total = {}'
+                                  .format(evt, n_events_tot))
+
+                        if n_events_tot >= nmax and nmax > -1:
+                            print('reached max nof of events (={})'
+                                  .format(nmax))
+                            break
```

Comment:
What is the point of the `nmax > -1` check? Can `nmax` ever be less than 0? I imagine that `nmax == 0` (or even `nmax >= 0` could be taken to mean 'unlimited number of events' in which case this meaning  needs to be mentioned in the docstring ... and in which case the check should be

    nmax > 0 and n_events_tot >= nmax

which is expressed more pythonically (but perhaps less clearly from the motivation perspective) as 

    n_events_tot >= nmax > 0